File: backend_ISA/src/main/java/com/bin448/backend/beans/Room.java
package com.bin448.backend.beans;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import javax.persistence.*;

@Entity
@Table(name = "room")
@JsonIgnoreProperties({ "hibernateLazyInitializer", "handler" })
public class Room {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    private Long number;
    @ManyToOne
    private Hotel hotel;
    @Column
    private Double pricePerNight;
    private Double avgGrade;

}

File: frontendISA/src/app/room/room.ts
export class Room {
  constructor(
    public number: number,
    public hotelId: number,
    public pricePerNight: number,
    public newPrice: number,
    public floor: number,
    public roomType: number,
    public avgGrade: number,
    public reserved: boolean
  ) {
  }
}
